src/pages/login/Login.tsx

```diff
@@ -1,3 +1,116 @@
+// import { useNavigate } from "react-router-dom";
+// import {
+//   Container,
+//   TextField,
+//   Button,
+//   Typography,
+//   Box,
+//   CircularProgress,
+// } from "@mui/material";
+// import { useAuth } from "../../context/AuthContext";
+
+// const Login: React.FC = () => {
+//   const [username, setUsername] = useState("");
+//   const [password, setPassword] = useState("");
+//   const [loading, setLoading] = useState(false);
+//   const [error, setError] = useState<string | null>(null);
+//   const navigate = useNavigate();
+//   const { login } = useAuth();
+
+//   const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+//     event.preventDefault();
+
+//     setLoading(true);
+//     setError(null);
+
+//     try {
+//       const res = await fetch("https://dummyjson.com/auth/login", {
+//         method: "POST",
+//         headers: { "Content-Type": "application/json" },
+//         body: JSON.stringify({
+//           username,
+//           password,
+//           expiresInMins: 30,
+//         }),
+//       });
+
+//       if (!res.ok) {
+//         throw new Error("Login failed");
+//       }
+
+//       const data = await res.json();
+
+//       login();
+
+//       navigate("/");
+//     } catch (err: any) {
+//       setError(err.message);
+//     } finally {
+//       setLoading(false);
+//     }
+//   };
+
+//   return (
+//     <Container
+//       component="main"
+//       maxWidth="xs"
+//       sx={{
+//         display: "flex",
+//         flexDirection: "column",
+//         alignItems: "center",
+//         mt: 8,
+//       }}
+//     >
+//       <Typography variant="h5" gutterBottom>
+//         Login
+//       </Typography>
+//       <Box
+//         component="form"
+//         onSubmit={handleSubmit}
+//         sx={{ mt: 1, width: "100%" }}
+//       >
+//         <TextField
+//           label="Username"
+//           variant="outlined"
+//           margin="normal"
+//           fullWidth
+//           required
+//           value={username}
+//           onChange={(e) => setUsername(e.target.value)}
+//         />
+//         <TextField
+//           label="Password"
+//           type="password"
+//           variant="outlined"
+//           margin="normal"
+//           fullWidth
+//           required
+//           value={password}
+//           onChange={(e) => setPassword(e.target.value)}
+//         />
+//         <Button
+//           type="submit"
+//           fullWidth
+//           variant="contained"
+//           color="primary"
+//           sx={{ mt: 2 }}
+//           disabled={loading}
+//         >
+//           {loading ? <CircularProgress size={24} color="inherit" /> : "Login"}
+//         </Button>
+//         {error && (
+//           <Typography color="error" sx={{ mt: 2 }}>
+//             {error}
+//           </Typography>
+//         )}
+//       </Box>
+//     </Container>
+//   );
+// };
+
+// export default Login;
+
```

Comment:
don't push unnecessary commented code, or comments. make the code looks clean and beautiful. :)